### PR TITLE
Add multiple dockerfile capability

### PR DESCRIPTION
**Pull request by dansku**
Change-type: minor
Signed-off-by: Daniel Andrade <daniel@balena.io>

### CODE REVIEW

**Comment by dimitrisnl** [comment]
Huh @resin-ci 	retest

File: pages/learn/develop/dockerfile.md [hunk 1]
```diff
@@ -101,6 +101,18 @@ Here are the supported machine names and architectures:
 
 {{> "general/deviceTypeNames"}}
 
+## Multiple Dockerfiles
+
+There are cases where just specifying a Dockerfile template doesn't work, as different systems or architectures may need different configuration and installation files. The solution to this problem is having multiple dockerfiles that are automatically selected based on the device it is going to be used.
+
+When using the command `balena push`, the balena CLI tool looks at all available dockerfiles and builds the appropriate container based on the following order:
```
Comment (pdcastro) [inline pages/learn/develop/dockerfile.md]: * Instead of _"builds the appropriate container",_ perhaps _"builds the appropriate docker image"._ (Because: a container is run out of a pre-built image; a container is a running instance of an image; an image is built.)
* Re _"the balena CLI tool looks at...", I am almost sure that:

Command | Where Dockerfile resolution and image building takes place
--- | ---
`git push` | All Dockerfiles are sent to the balenaCloud builder, where the appropriate Dockerfile is chosen.
`balena push` | All Dockerfiles are sent to the balenaCloud builder, where the appropriate Dockerfile is chosen.
`balena build` | The CLI chooses chooses the right file. The right file is sent to the specified docker daemon on a balena device or on a development machine (laptop or customer's own server), where the docker images are built.
`balena deploy` | When the `--build` option is given, it behaves like `balena build`. After the images are built, they are uploaded to balenaCloud.

* CLI version 9.14.5 and later is recommended as it fixes relevant issues.

File: pages/learn/develop/dockerfile.md [hunk 2]
```diff
@@ -101,6 +101,18 @@ Here are the supported machine names and architectures:
...
+
+* Dockerfile.<device-type>
+* Dockerfile.<arch>
+* Dockerfile.template
+
+As an example, if you have two dockerfiles in the directory, `Dockerfile.raspberrypi3` and `Dockerfile.template`, whenever you run `balena push <app-name>` if the `device-type` is a Raspberry Pi 3, it will select `Dockerfile.raspberrypi3`  and for all other devices it will automatically select `Dockerfile.template`.
```
Comment (pdcastro) [inline pages/learn/develop/dockerfile.md]: Perhaps better not to focus on `balena push` only, as the feature is available through `git push`, `balena push`, `balena build` and `balena deploy`.

File: pages/learn/develop/dockerfile.md [hunk 3]
```diff
@@ -101,6 +101,18 @@ Here are the supported machine names and architectures:
 
 {{> "general/deviceTypeNames"}}
 
+## Multiple Dockerfiles
+
+There are cases where just specifying a Dockerfile template doesn't work, as different systems or architectures may need different configuration and installation files. The solution to this problem is having multiple dockerfiles that are automatically selected based on the device it is going to be used.
```
Comment (CameronDiver) [inline pages/learn/develop/dockerfile.md]: > specifying a Dockerfile template doesn't work

sounds a bit weird to be and I don't think gives off the correct message.

I'd say something along the lines of;

There are cases when you would need a higher granularity of control when specifying build instructions for different devices and architectures than a Dockerfile template can provide. An example of this would be when different configuration or installation files are required.

File: pages/learn/develop/dockerfile.md [hunk 4]
```diff
@@ -101,6 +101,20 @@ Here are the supported machine names and architectures:
...
+
+There are cases when you would need a higher granularity of control when specifying build instructions for different devices and architectures than a Dockerfile template can provide. An example of this would be when different configuration or installation files are required.
+
+When deploying an application, the balena CLI tool looks at all available dockerfiles and builds the appropriate docker image based on the following order:
+
+* Dockerfile.<device-type>
+* Dockerfile.<arch>
+* Dockerfile.template
+
+As an example, if you have two dockerfiles in the directory, `Dockerfile.raspberrypi3` and `Dockerfile.template`, whenever you publish the application do balenaCloud, if the `device-type` is a Raspberry Pi 3, `Dockerfile.raspberrypi3` will be selected and for all other devices it will automatically select `Dockerfile.template`.
```
Comment (pdcastro) [inline pages/learn/develop/dockerfile.md]: "do balenaCloud" → "to balenaCloud"
The sentence got a bit long, maybe break it in too:
> As an example, say there are two dockerfiles in the directory, `Dockerfile.raspberrypi3` and `Dockerfile.template`. When the app is deployed to balenaCloud, if the `device-type` is a Raspberry Pi 3, `Dockerfile.raspberrypi3` will be selected and for all other devices it will automatically select `Dockerfile.template`.

File: pages/learn/develop/dockerfile.md [hunk 5]
```diff
@@ -101,6 +101,20 @@ Here are the supported machine names and architectures:
 
 {{> "general/deviceTypeNames"}}
 
+## Multiple Dockerfiles
+
+There are cases when you would need a higher granularity of control when specifying build instructions for different devices and architectures than a Dockerfile template can provide. An example of this would be when different configuration or installation files are required.
+
+When deploying an application, the balena CLI tool looks at all available dockerfiles and builds the appropriate docker image based on the following order:
```
Comment (pdcastro) [inline pages/learn/develop/dockerfile.md]: I know there is a note below clarifying that the feature works with `git push` as well, but I still think we could be more precise here, because this statement gives the impression that Dockerfile selection is a CLI feature. In the cases of 'git push' and 'balena push', it is not the CLI that looks at the available dockerfiles, it is the balenaCloud build servers. So maybe rephrase to _"When deploying an application, the balenaCloud build servers or the balena CLI tool look at all available dockerfiles ..."_
